tuf/ngclient/_internal/requests_fetcher.py

```diff
@@ -118,7 +120,7 @@ def _get_session(self, url: str) -> requests.Session:
         # reuse connections while minimizing subtle security issues.
         parsed_url = parse.urlparse(url)
 
-        if not parsed_url.scheme or not parsed_url.hostname:
+        if not parsed_url.scheme or (not parsed_url.hostname and parsed_url.scheme not in self.SCHEMES_WITHOUT_HOSTNAME):
```

Comment:
this is unlikely to pass our linter (line looks too long).

That said, I wonder if the check is even useful? IIRC it was added just to replace some earlier URL sanity checks, and it's certainly not "real" URL validation: The intent is only to make sure the session index lookup a few lines later works as intended and it seems to me that it does... I think removing the host name check (or even the whole if-clause) may be fine.